### PR TITLE
[mlir][vector] Update tests/patterns for vector.transpose

### PR DESCRIPTION
**Pull request by banach-space**
Pretty much all logic that we have today for lowering vector.transpose
assumes fixed length vectors (it's done via vector.shuffle that don't
support scalable vectors). This patch updates related tests and patterns
to capture and document that limitation more explicitly.

Note that `vector.transpose` is a valid operation in the context of
scalable vectors, but we are yet to implement the missing lowerings.

Most changes are implemented in the test file. Here's a summary:
* @transpose_nx8x2xf32 is renamed as @transpose_scalable and move near
  other test using `lowering_strategy = "shuffle_1d"` (to avoid
  duplicating TD sequences)
* tests specific to X86  (`avx2_lowering_strategy = true`) are moved to
  a dedicated file (to seperate generic tests from target-specific
  tests)
* `@transpose10_nx4xnx1xf32` duplicated `@transpose10_4xnx1xf32` and was
  deleted (the latter is renamed as `@transpose10_4x1xf32_scalable` to
  match its fixed-width counterpart: `@transpose10_4x1xf32`)

### CODE REVIEW

**Comment by hanhanW** [comment]
> tests specific to X86 (avx2_lowering_strategy = true) are moved to a dedicated file (to seperate generic tests from target-specific tests)

Where is the file? I don't see it in the PR. It looks like you're removing bunch of lit tests from `vector-transpose-lowering.mlir`.

**Comment by banach-space** [comment]
> > tests specific to X86 (avx2_lowering_strategy = true) are moved to a dedicated file (to seperate generic tests from target-specific tests)
> 
> Where is the file? I don't see it in the PR. It looks like you're removing bunch of lit tests from `vector-transpose-lowering.mlir`.

Sorry about that, forgot to attach it. Fixed in the latest commit.

File: mlir/test/Dialect/Vector/CPU/X86/vector-transpose-lowering.mlir [hunk 1]
```diff
@@ -0,0 +1,492 @@
+// RUN: mlir-opt %s --transform-interpreter --split-input-file | FileCheck %s
+
+// NOTE: This file tests lowering from the X86 dialect. Since X86 does not
+// support scalable vectors, all examples in this file use fixed-width vectors.
```
Comment (MacDue) [inline mlir/test/Dialect/Vector/CPU/X86/vector-transpose-lowering.mlir:4]: This is not lowering from the x86 dialect (also to be clear there's no x86 dialect, only x86vector), the input is the vector dialect. However, this does not actually lower to x86vector dialect operations, it lowers directly to inline assembly.

File: mlir/test/Dialect/Vector/CPU/X86/vector-transpose-lowering.mlir [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+
+// CHECK-LABEL: func @transpose4x8
+func.func @transpose4x8xf32(%arg0: vector<4x8xf32>) -> vector<8x4xf32> {
```
Comment (MacDue) [inline mlir/test/Dialect/Vector/CPU/X86/vector-transpose-lowering.mlir:7]: Note: I diffed these tests with the deleted tests -- identical match.

**Review by MacDue — APPROVED** [review]
Minor nit, otherwise LGTM.

**Review by hanhanW — APPROVED** [review]

**Pull request merged**